porebuilder/porebuilder.py

```diff
@@ -108,3 +111,89 @@ def __init__(self, pore_depth=4, side_dim=3, n_sheets=3, pore_width=1,
             self.add(mb.clone(child))
 
         self.periodicity = box.maxs
+
+
+class GraphenePoreFunctionalized(mb.Compound):
+    """A general slit pore recipe that functionalizes the inner surfaces
+
+    Parameters
+    ----------
+    pore_depth : int, default=4
+        dimensions of graphene sheet in x direction in nm
+    side_dim : int, default=4
+        dimensions of graphene sheet in z direction in nm
+    n_sheets : int, default=3
+        number of parallel graphene sheets
+    pore_width: int, default=1
+        width of slit pore in nm
+    func_groups : mb.Compound or List-like thereof, defualt=None
+        compounds to attach to the surface of a slit pore
+    func_ports : string or List-like thereof, defualt='up'
+        ports directly accesible by the func_groups that are the point of
+         attachment to the pore
+    func_percent : float or List-like thereof, defualt=.03
+        percentage of surface to be functionalized by groups in func_groups
+    NOTE: Both func_ports and func_percent must be in the same order as their
+        corresponding functional group in func_groups. If only one value is
+        given for either func_ports or func_percent that value will be used for all
+        compounds in func_groups.
+    Attributes
+    ----------
+    see mbuild.Compound
+
+    """
+
+    def __init__(self, pore_depth=4, side_dim=3, n_sheets=3, pore_width=1,
+                 func_groups=None, func_percent=.03, func_ports='up'):
+
+        super(GraphenePoreFunctionalized, self).__init__()
+
+        pore = GraphenePore(pore_depth=pore_depth, side_dim=side_dim,
+                            n_sheets=n_sheets, pore_width=pore_width)
+
+        if isinstance(func_groups, mb.Compound):
+            func_groups = [func_groups]
+        if isinstance(func_percent, float):
+            func_percent = [func_percent] * len(func_groups)
+        if isinstance(func_ports, string_types):
+            func_ports = [func_ports] * len(func_groups)
+
+        if len(func_groups) != len(func_percent) or len(
+                func_groups) != len(func_ports):
+            raise ValueError(
+                "If more than one port name or percent is given then "
+                "it must be specifeid for all functional groups")
+
+        Top = pore.children[0]
+        Bot = pore.children[1]
+
+        t_surface = []
+        b_surface = []
+
+        for C in Top:
+            if C.pos[0] >= 0 and C.pos[1] <= .336 * (n_sheets - 1) + pore_width:
+                t_surface.append(C)
+        
+        for C in Bot:
+            if C.pos[0] >= 0 and C.pos[1] >= .335 * (n_sheets - 1):
+                b_surface.append(C),
+
+        for side, surface in zip((Top,Bot),(t_surface,b_surface)):
```

Comment:
Can we raname `side` to something like `pore_wall`? It clashes a little bit with the `side_dim` argument we are using elsewhere.